src/gateway/environments/jsdom-sixteen/jsdom-sixteen-resource-loader.js

```diff
@@ -0,0 +1,144 @@
+// @flow
+import {URL} from "url";
+import {ResourceLoader} from "jsdom";
+import type {FetchOptions} from "jsdom";
+import type {RequestOptions, RenderAPI} from "../../types.js";
+import {getAgentForURL} from "../../../shared/index.js";
+import {
+    DefaultRequestOptions,
+    request,
+    abortInFlightRequests,
+} from "../../request.js";
+import {applyAbortablePromisesPatch} from "./apply-abortable-promises-patch.js";
+
+/**
+ * A ResourceLoader implementation for JSDOM sixteen-compatible versions of
+ * JSDOM that only allows for fetching JS files.
+ *
+ * A JS file request is identified by the regular expression:
+ *   /^.*\.js(?:\?.*)?/g
+ */
+export class JSDOMSixteenResourceLoader extends ResourceLoader {
+    /**
+     * Used to indicate if any pending requests are still needed so that we
+     * can report when an unused request is fulfilled.
+     */
+    _active: boolean;
+    _renderAPI: RenderAPI;
+    _requestOptions: RequestOptions;
+
+    static get EMPTY_RESPONSE(): Promise<Buffer> {
+        return Promise.resolve(Buffer.from(""));
+    }
```

Comment:
This is now an accessor so that it creates a new value on call. That way, it will definitely get an abortable promise.